macOS/DuckDuckGo/Updates/UpdateController.swift

```diff
@@ -186,12 +213,33 @@ final class UpdateController: NSObject, UpdateControllerProtocol {
 
         Logger.updates.log("Checking for updates respecting rollout")
 
+        userDriver?.userCheckedForUpdates()
         updater.checkForUpdatesInBackground()
     }
 
     // Check for updates immediately, bypassing the rollout schedule
     // This is used for user-initiated update checks only
     func checkForUpdateSkippingRollout() {
+        let discardCurrentUpdate = !autoRestartAllowed && shouldForceUpdateCheck
```

Comment:
This is the new expiration logic.  This will be true if the existing update must be discarded.